src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete state[keys];
+        }
+        break;
+
+      default:
+        for (const keys in state) {
+          delete state[keys];
+        }
+        break;
```

Comment:
```suggestion
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;
```

Reply:
why important: in this task, you're given a clear explanation of the different cases, and here's a slight amount of code, but when it comes to a larger project, other devs may get stuck understanding what's written

Reply:
also, don't forget to use the single form as you take out only one item, not a bunch